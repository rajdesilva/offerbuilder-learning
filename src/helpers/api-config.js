import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});