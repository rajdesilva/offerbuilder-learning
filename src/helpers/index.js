export * from './api-config';
export * from './history';
