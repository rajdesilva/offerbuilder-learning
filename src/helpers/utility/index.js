export { createSearchQueryURL } from './createSearchQueryURL';
export { isParentOrChildrenPropertyAdded } from './isParentOrChildrenPropertyAdded';
export { isPropertySelected } from './isPropertySelected';
export { mapBrands } from './mapBrands';
export { mapOnlySuppliers } from './mapOnlySuppliers';
export { parseFormDescriptionsAndImagesForOffer } from './parseFormDescriptionsAndImagesForOffer';
export { removePropertyForSelectedChannels } from './removePropertyForSelectedChannels';
export { removePropertyForSelectedSuppliers } from './removePropertyForSelectedSuppliers';
export { getParsedProperty } from './getParsedProperty';
export { getFormattedValue } from './getFormattedValue';
export { getValue } from './getValue';
export { parsePropertyMarketingInfoToFormData } from './parsePropertyMarketingInfoToFormData';
export { createOfferRequestObject } from './createOfferRequestObject';
export { getParsedOffers } from './getParsedOffers';
export { getListOfPropertiesForSupplier } from './getListOfPropertiesForSupplier';
export { getListOfPropertiesForChannels } from './getListOfPropertiesForChannels';
export { getInitializedState } from './getInitializedState';
export { getParsedNewOfferSettingsObject } from './getParsedNewOfferSettingsObject';
export { getListOfPropertiesForStorefront } from './getListOfPropertiesForStorefront';
export { checkIfMultipleChannelPresentForTarget } from './checkIfMultipleChannelPresentForTarget';
export { checkIfMultipleSupplierPresentForTarget } from './checkIfMultipleSupplierPresentForTarget';
export { checkIfMultipleStorefrontPresentForTarget } from './checkIfMultipleStorefrontPresentForTarget';
export { removePropertyForSelectedStorefront } from './removePropertyForSelectedStorefront';
export { checkIfMultipleBrandPresentForTarget } from './checkIfMultipleBrandPresentForTarget';
export { getListOfPropertiesForBrand } from './getListOfPropertiesForBrand';
export { getSearchStateFromRedux } from './getSearchStateFromRedux';
export { getObjectStringify } from './getObjectStringify';
export { parseOfferMarketingFromOffer } from './parseOfferMarketingFromOffer';
export { parseOfferMarketingLanguagesFromOffer } from './parseOfferMarketingLanguagesFromOffer';
export { getFilterData } from './getFilterData';
export { getSearchAndFilterQueryURL } from './getSearchAndFilterQueryURL';
export { getSearchQueryURLForEditAddProperties } from "./getSearchQueryURLForEditAddProperties";
export { isSavedOfferProperty } from './isSavedOfferProperty';
export { getParsedPropertiesForDeeplinkSettings } from "./getParsedPropertiesForDeeplinkSettings";
export { resetApplicationReduxAndStorage } from "./resetApplicationReduxAndStorage";
export { updateSettingsBrandWithSearchedValues } from "./updateSettingsBrandWithSearchedValues";
export { checkIfUserHasRole } from "./checkIfUserHasRole";
export { getUserOfferBuilderRole } from "./getUserOfferBuilderRole";
export { checkIfNewPropertiesAdded } from "./checkIfNewPropertiesAdded";
export { getAllBrandsWithoutKeysFromTempBrands } from "./getAllBrandsWithoutKeysFromTempBrands";
export { checkIfItemAlreadyPresentInList } from "./checkIfItemAlreadyPresentInList";
export { checkIfTravellingDateValidAsPerBookingDate } from "./checkIfTravellingDateValidAsPerBookingDate";
export { checkIfPropertiesHaveMultipleLocations } from "./checkIfPropertiesHaveMultipleLocations";
export { displayMargin } from "./displayMargin";
export { getFilterCount } from "./getFilterCount";
export { labelForBrands } from "./labelForBrands";
export { getDistanceFromLatLonInKm } from "./getDistanceFromLatLonInKm";
export { getTrustYouIdListFromProperties } from "./getTrustYouIdListFromProperties";
export { getSearchParamsToFetchMarketPrice } from "./getSearchParamsToFetchMarketPrice";
export { getMarketPriceForPropertyFromList } from "./getMarketPriceForPropertyFromList";
export { getImageFilterCount } from "./getImageFilterCount";
export { getSearchAndFilterImagesQueryUrl } from "./getSearchAndFilterImagesQueryUrl";
export { getImageName } from "./getImageName";
export { updateReduxWithOfferData } from "./updateReduxWithOfferData";
export { getOfferTypeFromPropertiesSelected } from "./getOfferTypeFromPropertiesSelected";