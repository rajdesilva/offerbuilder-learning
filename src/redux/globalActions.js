export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}