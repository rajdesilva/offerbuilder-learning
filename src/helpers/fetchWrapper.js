import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
