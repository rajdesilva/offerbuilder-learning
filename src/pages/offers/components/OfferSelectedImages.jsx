import React, { Fragment } from "react";
import { Image, List, Row, Button, Col } from "antd";
import { FormattedMessage } from "react-intl";
import defaultImg from "../../../assets/images/default-img.svg";
import { CloseOutlined } from "@ant-design/icons";
import styles from "./css/OfferSelectedImages.module.less";
import { getImageName } from "../../../helpers/utility";

export default function OfferSelectedImages(props) {
  const removeImageFromIndex = (removedIndex) => {
    const updatedArray = props.selectedImages.filter(
      (word, index) => index !== removedIndex
    );
    props.setUpdatedSelectedImages(updatedArray);
  };

  return (
    <div
      className={styles["selected-area"]}
      data-testid="selected-images-section"
    >
      <Row justify="space-between">
        <Col>
          <FormattedMessage id="nemo.selectedImages" />:
        </Col>
        <Col>
          <Button
            type="link"
            data-testid="deselect-image-all-btn"
            onClick={() => props.setUpdatedSelectedImages([])}
          >
            <FormattedMessage id="nemo.deselectAll" />
          </Button>
        </Col>
      </Row>
      <Image.PreviewGroup>
        <List
          grid={{ gutter: 14, column: props.isDetailsShown ? 7 : 10 }}
          dataSource={props.selectedImages}
          className={styles["list-item-img"]}
          renderItem={(image, index) => {
            return (
              <List.Item
                key={index + "-" + image.id}
                data-testid={index + "-" + image.id + "-image-list-item"}
              >
                <Image
                  src={image.url}
                  className={styles.imgSelected}
                  preview={false}
                  onClick={() => props.setCarouselImage(index, true, true)}
                  onError={(ev) => (ev.target.src = defaultImg)}
                ></Image>
                <Button
                  data-testid={index + "-remove-image-btn"}
                  ghost
                  className={styles["remove-image-btn"]}
                  type="text"
                  icon={<CloseOutlined className={styles.removeIcon} />}
                  onClick={() => removeImageFromIndex(index)}
                ></Button>
              </List.Item>
            );
          }}
        ></List>
      </Image.PreviewGroup>
      <Row>
        <div
          data-testid="selected-images-name-list"
          className={styles.imgTextDiv}
        >
          <span className={styles.imgsUsedText}>
            <FormattedMessage id="nemo.theseImagesWillBeUsedInOffer" />{" "}
          </span>{" "}
          :
          {props.selectedImages.map((image, index) => {
            return (
              <Fragment>
                {index === 0 ? "" : ","}
                <Button
                  size="small"
                  type="link"
                  data-testid={index + "-selected-image-name"}
                  onClick={() => {
                    props.displayImageDetails(image, index, true);
                  }}
                  className={styles.imgNameBtn}
                >
                  {getImageName(image.url) || (
                    <FormattedMessage id="nemo.unknown" />
                  )}
                </Button>
              </Fragment>
            );
          })}
        </div>
      </Row>
    </div>
  );
}
