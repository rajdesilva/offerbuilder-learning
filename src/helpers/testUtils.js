import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
