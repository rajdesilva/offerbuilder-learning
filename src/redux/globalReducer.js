import { globalActions } from "./globalActions";
import produce from "immer";

const initialState = {
  error: "",
  success: "",
  language: "",
};
export const globalReducer = produce((draft, action) => {
  switch (action.type) {
    case globalActions.SET_ERROR:
      draft.success = "";
      draft.error = action.payload;
      return;
    case globalActions.SET_SUCCESS:
      draft.error = "";
      draft.success = action.payload;
      return;
    case globalActions.SET_LANGUAGE:
      draft.language = localStorage.getItem('language')
      return;
    case globalActions.CLEAR:
      return initialState;
    default:
      return draft;
  }
}, initialState);
