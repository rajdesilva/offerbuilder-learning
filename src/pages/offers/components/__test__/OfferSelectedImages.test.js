import React from "react";
import { createStore } from "redux";
import { render, fireEvent } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import OfferSelectedImages from "../OfferSelectedImages";
import { tempStore } from "./tempStore";

describe("OfferSelectedImages component test", () => {
  test("create OfferSelectedImages components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      setUpdatedSelectedImages: jest.fn(),
      isDetailsShown: false,
      displayImageDetails: jest.fn(),
      selectedImages: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 1-30-49 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
          offerDetails: [],
        },
      ],
    };
    const { container } = render(
      <MemoryRouter>
        <OfferSelectedImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(container.firstChild).toMatchSnapshot();
  });
  test("OfferSelectedImages components- click on deselect all to remove all images", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      setUpdatedSelectedImages: jest.fn(),
      isDetailsShown: false,
      displayImageDetails: jest.fn(),
      selectedImages: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 1-30-49 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
          offerDetails: [],
        },
      ],
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferSelectedImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const deselectBtn = getByTestId("deselect-image-all-btn");
    fireEvent.click(deselectBtn);
    expect(props.setUpdatedSelectedImages).toHaveBeenCalledTimes(1);
  });
  test("OfferSelectedImages components- click on remove image to remove 1st selected image", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      setUpdatedSelectedImages: jest.fn(),
      isDetailsShown: false,
      displayImageDetails: jest.fn(),
      selectedImages: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 1-30-49 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
          offerDetails: [],
        },
      ],
    };

    const { getByTestId } = render(
      <MemoryRouter>
        <OfferSelectedImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const removeBtn = getByTestId("0-remove-image-btn");
    fireEvent.click(removeBtn);
    expect(props.setUpdatedSelectedImages).toHaveBeenCalledTimes(1);
  });

  test("OfferSelectedImages components- Check if user click on image name to see image details", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      setUpdatedSelectedImages: jest.fn(),
      isDetailsShown: false,
      displayImageDetails: jest.fn(),
      selectedImages: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 1-30-49 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
          offerDetails: [],
        },
      ],
    };
    const { queryByTestId, getByTestId } = render(
      <MemoryRouter>
        <OfferSelectedImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(queryByTestId("0-selected-image-name")).toBeInTheDocument();
    const nameBtn = getByTestId("0-selected-image-name");
    expect(nameBtn).toBeTruthy();
    expect(nameBtn).toBeInTheDocument();
    fireEvent.click(nameBtn);
    expect(props.displayImageDetails).toHaveBeenCalledTimes(1);
  });
});
