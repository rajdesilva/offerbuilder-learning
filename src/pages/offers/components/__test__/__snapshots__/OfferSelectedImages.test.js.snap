// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferSelectedImages component test create OfferSelectedImages components snapshot 1`] = `
<div
  data-testid="selected-images-section"
>
  <div
    class="ant-row ant-row-space-between"
  >
    <div
      class="ant-col"
    >
      Selected images
      :
    </div>
    <div
      class="ant-col"
    >
      <button
        class="ant-btn ant-btn-link"
        data-testid="deselect-image-all-btn"
        type="button"
      >
        Deselect all
      </button>
    </div>
  </div>
  <div
    class="ant-list ant-list-split ant-list-grid"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-row"
          style="margin-left: -7px; margin-right: -7px;"
        >
          <div
            style="width: 10%; max-width: 10%;"
          >
            <div
              class="ant-col"
              style="padding-left: 7px; padding-right: 7px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="0-Screenshot 2021-03-30 at 1-14-53 PM.png-image-list-item"
              >
                <div
                  class="ant-image"
                >
                  <img
                    class="ant-image-img"
                    src="https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png"
                  />
                </div>
                <button
                  class="ant-btn ant-btn-text ant-btn-icon-only"
                  data-testid="0-remove-image-btn"
                  type="button"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            style="width: 10%; max-width: 10%;"
          >
            <div
              class="ant-col"
              style="padding-left: 7px; padding-right: 7px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="1-Screenshot 2021-03-24 at 2-46-07 PM.png-image-list-item"
              >
                <div
                  class="ant-image"
                >
                  <img
                    class="ant-image-img"
                    src="https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png"
                  />
                </div>
                <button
                  class="ant-btn ant-btn-text ant-btn-icon-only"
                  data-testid="1-remove-image-btn"
                  type="button"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            style="width: 10%; max-width: 10%;"
          >
            <div
              class="ant-col"
              style="padding-left: 7px; padding-right: 7px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="2-Screenshot 2021-03-24 at 1-30-49 PM.png-image-list-item"
              >
                <div
                  class="ant-image"
                >
                  <img
                    class="ant-image-img"
                    src="https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png"
                  />
                </div>
                <button
                  class="ant-btn ant-btn-text ant-btn-icon-only"
                  data-testid="2-remove-image-btn"
                  type="button"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      data-testid="selected-images-name-list"
    >
      <span>
        These images already exist in the library and will be used in offer
         
      </span>
       
      :
      
      <button
        class="ant-btn ant-btn-link ant-btn-sm"
        data-testid="0-selected-image-name"
        type="button"
      >
        <span>
          Screenshot%202021-03-30%20at%201-14-53%20PM.png
        </span>
      </button>
      ,
      <button
        class="ant-btn ant-btn-link ant-btn-sm"
        data-testid="1-selected-image-name"
        type="button"
      >
        <span>
          Screenshot%202021-03-24%20at%202-46-07%20PM.png
        </span>
      </button>
      ,
      <button
        class="ant-btn ant-btn-link ant-btn-sm"
        data-testid="2-selected-image-name"
        type="button"
      >
        <span>
          Screenshot%202021-03-24%20at%201-30-49%20PM.png
        </span>
      </button>
    </div>
  </div>
</div>
`;
